### PR TITLE
test: set up testing environment of jest and add test

### LINKED ISSUE #63: util/initialize.tsのテスト作成
cosmiconfigをモックするかは要検討

### LINKED ISSUE #65: テスト環境の構築
無難にjestで良さそう。
完璧なテストを行うなら、jestではなくshellspecあたりになってくると思う

### PR DESCRIPTION
**Pull request by camcam-lemon**
Close #65 
Close #63 

## Done
- Create a test environment with jest
- Create test file for initialize.ts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,17 +33,20 @@
     "commit": "git-cz",
     "dev": "tsc -w",
     "release": "shipjs prepare",
+    "test": "jest",
+    "test:watch": "jest --watch",
     "test:cli": "sh scripts/test-cli",
     "typecheck": "tsc --project ./tsconfig.build.json --noEmit"
   },
   "dependencies": {
-    "@types/inquirer": "^7.3.0",
```
[the remark this answers is not in the document]
Comment (camcam-lemon) [reply package.json:40]: ok, understood
I undo change.

File: __test__/util/ initialize.spec.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  }));
+
+  return {
+    cosmiconfigSync,
+  };
+});
+
+describe("util/initialize", () => {
+  describe("initialize()", () => {
+    it("configプロパティにcosmiconfigのconfig値が返ってくるか", () => {
```
[the remark this answers is not in the document]
Comment (camcam-lemon) [reply __test__/util/ initialize.spec.ts:28]: Do my best😂

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  preset: "ts-jest",
+  globals: {
+    "ts-jest": {
+      tsConfig: "./tsconfig.test.json",
+    },
+  },
+  transform: {
+    "^.+\\.ts$": "ts-jest",
+  },
+  testRegex: "(/__tests__/.*|(\\.|/)(test|spec))\\.tsx?$",
```
[the remark this answers is not in the document]
Comment (camcam-lemon) [reply jest.config.js:11]: sorry...

**Comment by tyankatsu0105** [comment]
@camcam-lemon 
Probably, `__test__/util/ initialize.spec.ts` is include space at first letter

**Comment by camcam-lemon** [comment]
> Could you add the GitHub action workflow's file that running tests with various node versions?
> Like this: https://github.com/tyankatsu0105/eslint-plugin-config-files/blob/master/.github/workflows/test.yml

OK.
However, it will be another issue.
Because... I have never written a github action.
I need time to learn.

> Which version do you think we should support?
> IMO: 10, 12, 14 https://nodejs.org/ja/about/releases/

- node10...if possible
- node12...must
- node14...must

Node10 is maintenance mode.
I think node10 Included in the engines fields, but `not much` support.
How about?

**Comment by camcam-lemon** [comment]
> Probably, `__test__/util/ initialize.spec.ts` is include space at first letter

understood.
It's a suggestion, can I rename the folder  to `test` or `tests`?
`__test__`  is not common.

**Comment by tyankatsu0105** [comment]
>However, it will be another issue.

I agree.
Let's create an issue🧑‍💻

>I have never written a github action.

GitHub Action's syntax is simple. I think that you'll understand soon.
[This extension](https://marketplace.visualstudio.com/items?itemName=me-dutour-mathieu.vscode-github-actions) is very useful.

**Comment by tyankatsu0105** [comment]
About the version of node, we should support version 10.
Because EoL has not been coming yet.

**Comment by tyankatsu0105** [comment]
>can I rename the folder to test or tests?

Sure👍 You can do it.

**Comment by camcam-lemon** [comment]
> I agree.
> Let's create an issue🧑‍💻

Created an issue #69 

> GitHub Action's syntax is simple. I think that you'll understand soon.
> [This extension](https://marketplace.visualstudio.com/items?itemName=me-dutour-mathieu.vscode-github-actions) is very useful.

Thunks!
VSCode extenstion is great!!

**Comment by camcam-lemon** [comment]
> About the version of node, we should support version 10.
> Because EoL has not been coming yet.

I agree👍🏻

Do the following👇
- [] add package.json engine =>  node10, node12, node14
- [] delete leading white space to `initialize.spec.ts`
- [] rename __test__ => tests

I chose tests because your other repository testing file name is tests💁🏻‍♀️

File: package.json [hunk 4]
```diff
@@ -52,13 +55,15 @@
     "eslint": "^7.6.0",
     "eslint-config-prettier": "6.11.0",
     "eslint-plugin-prettier": "3.1.4",
+    "jest": "^26.2.2",
     "lint-staged": "10.2.11",
     "prettier": "2.0.5",
     "shipjs": "0.20.1",
+    "ts-jest": "^26.1.4",
```
Comment (tyankatsu0105) [inline package.json:62]: Nice choice.
ts-jest is good tools👍

**Pull request merged**